### ISSUE TITLE
Macros not defined across multiple files.

### ISSUE THREAD

**Issue by mforbes** [mcmtroffaes/sphinxcontrib-bibtex#261]
With BibTeX, one can define macros in one file, then use them in another.  For example, the following works:

```bibtex
@Comment{local.bib}
@string{rmp = "Rev. Mod. Phys."}
@article{Toussaint:2011,
	author = {von Toussaint, Udo},
	doi = {10.1103/RevModPhys.83.943},
	issue = {3},
	journal = rmp,
	month = sep,
	numpages = {0},
	pages = {943--999},
	publisher = {American Physical Society},
	title = {Bayesian inference in physics},
	volume = {83},
	year = {2011},}
```

But if I remove the `@string{rmp = "Rev. Mod. Phys."}` and place it in another file `macros.bib`, then

```python
# conf.py
...
bibtex_bibfiles = ["macros.bib", "local.bib"]
...
```

fails with the following error:

```bash
$ make html
Running Sphinx v3.5.4
checking for .../macros.bib in bibtex cache... not found
parsing bibtex file .../macros.bib... parsed 0 entries
checking for .../local.bib in bibtex cache... not found
parsing bibtex file .../local.bib... 
Exception occurred:
  File ".../python3.8/site-packages/pybtex/errors.py", line 78, in report_error
    raise exception
pybtex.database.input.bibtex.UndefinedMacro: undefined string in line 6: rmp
...
```

**Comment by mcmtroffaes** [comment]
Yes, currently, bib files are parsed independently. Not sure how to address this... I'll need to have a look if pybtex can support this scenario.

**Comment by mforbes** [comment]
Thanks for the confirmation. Somehow the state should be maintained.  This is what is done with BibTeX and LaTeX in general.  

Of course, the easy workaround is to concatenate the files.  It seems like there are no issues with duplicate macro definitions, so it might even be possible to do this in python before passing the files to `pybtex`... but I have not explored this much yet.

**Comment by mcmtroffaes** [comment]
Fixed in #262.
